test/specs/uriParameters.js

```diff
@@ -0,0 +1,140 @@
+      ].join('\n'), 'api.raml')
+        .then(function (result) {
+          setTimeout(function () {
+            var expected =   {
+              title: 'Title',
+              baseUri: 'http://domain/{+l2Parameter}',
+              baseUriParameters: {
+                l2Parameter: { type: 'number', displayName: 'l2Parameter', required: true },
+              },
+              protocols: [ 'HTTP' ]
+            };
+
+            result.should.deep.equal(expected);
+            done();
+          }, 0);
+        });
+    });
+
+    it('in uriParameters', function (done) {
+      return raml.load([
+        '#%RAML 0.8',
+        'title: Title',
+        'baseUri: http://domain',
+        '/resource/{+newParameter}:',
+        '  uriParameters:',
+        '    newParameter:',
+        '      type: number'
+      ].join('\n'), 'api.raml')
+        .then(function (result) {
+          setTimeout(function () {
+            var expected =   {
+              title: 'Title',
+              baseUri: 'http://domain',
+              protocols: [ 'HTTP' ],              
+              resources: [{
+                uriParameters: {
+                  newParameter: { type: 'number', displayName: 'newParameter', required: true },
+                },
+                "relativeUri": "/resource/{+newParameter}",
+                "relativeUriPathSegments": [
+                  "resource",
+                  "{+newParameter}"
+                ]
+              }]
+            };
+
+            result.should.deep.equal(expected);
+            done();
+          }, 0);
+        }, function (err) {console.log(err)});
+    });
+  });
+
+  describe('more than one parameter in bracket', function () {
+    it('in baseUriParameters', function (done) {
+      return raml.load([
+        '#%RAML 0.8',
+        'title: Title',
+        'baseUri: http://domain/{parameter1,parameter2}/{parameter3,parameter4}'
+      ].join('\n'), 'api.raml')
+        .then(function (result) {
+          setTimeout(function () {
+            var expected =   {
+              title: 'Title',
+              baseUri: 'http://domain/{parameter1,parameter2}/{parameter3,parameter4}',
+              protocols: [ 'HTTP' ],
+              baseUriParameters: {
+                parameter1: { type: 'string', displayName: 'parameter1', required: true },
+                parameter2: { type: 'string', displayName: 'parameter2', required: true },
+                parameter3: { type: 'string', displayName: 'parameter3', required: true },
+                parameter4: { type: 'string', displayName: 'parameter4', required: true }
+              },
+            };
+
+            result.should.deep.equal(expected);
+            done();
+          }, 0);
+        }, console.log);
+    });
+
+    it('in uriParameters', function (done) {
+      return raml.load([
+        '#%RAML 0.8',
+        'title: Title',
+        'baseUri: http://domain',
+        '/resource/{parameter1,parameter2}:'
```

Comment:
Neither, can we just skip these tests and keep the implementation to level 2 only? The parser was already used for just level 1, so I'm guessing it's simple enough? 

If it helps, other tooling I'll be implementing will literally be based on RegExps instead of the parser.